engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala

```diff
@@ -186,13 +188,29 @@ final class SuggestionsHandler(
 
     case msg: Api.SuggestionsDatabaseSuggestionsLoadedNotification
         if state.isSuggestionLoadingRunning =>
-      state.suggestionLoadingQueue.enqueue(msg)
+      logger.trace(
+        "SuggestionsDatabaseSuggestionsLoadedNotification [shouldStartBackgroundProcessing={}].",
+        state.shouldStartBackgroundProcessing
+      )
+      if (state.shouldStartBackgroundProcessing) {
```

Comment:
So this was the main source of the problem?

Reply:
I had to properly synchronize the state of background notifications in runtime
https://github.com/enso-org/enso/blob/35fdcd2c07f646dc1f44522c5b21a6e17c155a81/engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala#L317-L326